Source_Code/TM_Backend/TeamMentor.Users/Users/Users_Creation.cs

```diff
@@ -192,6 +192,16 @@ public static Signup_Result createTmUserResponse(this TM_UserData userData, NewU
             }
             // ensure the email is lowercase (will fail validation otherwise)
             newUser.Email = newUser.Email.lower();
+            //Email check (length, null, valid)
+            if (newUser.Email.isNull() || newUser.Email.Length>256 || !IsValidEmailAddress(newUser.Email))
+            {
+                if (newUser.Email.notNull() && newUser.Email.Length> 256)

```

Comment:
this if sequence is confusing, @michaelhidalgo can you clean this?

Also use the valid_Extension_Method  (in fact you could add an **not_Email_Address** which might make it more readable)

Btw, is there an **bigger_Than** FluentSharp extension method? (if not can you add it to the _Extra_Extension_Methods.cs file?), that way you can use

```csharp
newUser.Email.bigger_Than(256)
```